src/app/teams/teams-view.component.ts

```diff
@@ -362,4 +362,12 @@ export class TeamsViewComponent implements OnInit, OnDestroy {
     ).subscribe(() => this.planetMessageService.showMessage(`${member.name} has been made Leader`));
   }
 
+  removeCourse(item) {
+    const index = this.team.courses.indexOf(item);
+    this.team.courses.splice(index, 1);
```

Comment:
I should have gone with my hunch earlier, but this will cause problems if there is an error updating the team with the removed course.

`Array.splice` is a mutable function, so it changes `this.team` before the change is made to the database.  If there's an error, the page will update showing the course removed even though it still exists as part of the team.

[This blog post](https://flaviocopes.com/how-to-remove-item-from-array/) is a good summary of the ways to remove an item from an array in an immutable manner.  Either of the first two options will work, and you can add the new courses array directly to the object being sent to `updateTeam`:
```
this.teamsService.updateTeam({ ...this.team, courses: newCourses })
```

Reply:
Hey Paul, I will work on these changes. Thank you for the feedback.